source/faq.txt

```diff
@@ -4,12 +4,176 @@ FAQ
+   
+   With the entity listed above, you can mark the ``id`` field to encode
+   as ``_id``:
+
+   .. code-block:: java
+
+      @BsonId()
+      public final ObjectId id;
+
+Can the ID field be a compound key?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes. For an example of this, see `our implementation <https://github.com/niccottrell/mongo-java-tests/blob/master/src/test/PojoCompoundIdTest.java>`_
+
+What is the discriminator?
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The discriminator is for cases where you want to use inheritance, and
+store multiple types of documents within the same collection or parent
+document (in case you embed sub-documents). 
+
+For example, if you have an ``Event`` class, that you extend in Java (e.g.
+``MachineEvent`` or ``NetworkEvent``), using the discriminator identifies
+which class the PojoCodec should use to serialize/deserialize the
+document. 
+
+Can I control serialization of ``LocalDate``?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes, the 3.7 Java driver adds native support for ``JSR-310 Instant``,
+``LocalDate`` & ``LocalDateTime``. 
+
+Add the Codec to the top your registry before the ``DefaultCodecRegister``
+and before the ``PojoCodecProvider``: 
+
+.. code-block:: java
+
+   static final CodecRegistry CODEC_REGISTRY = fromRegistries(
+      CodecRegistries.fromCodecs(
+                        new MyDateAsStringCodec()),
+                        MongoClient.getDefaultCodecRegistry(),
+                        fromProviders(pojoCodecProvider));
+
+Can I serialize a ``java.util.Date`` as a string in format **yyyy-mm-dd**?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Yes, you can build your own Codec for this class and add it to the registry. 
+
+Can I make POJOs read/write directly to the field and not use the getters/setters at all?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Not natively, but you can build your own POJO Codec.
+
+However, the ``SET_PRIVATE_FIELDS_CONVENTION`` sets a private field via
+reflection if no public setter is available. 
+
+Can I mix private, protected and public setters and getters?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+No. The native POJO Codec assumes that getters/setters have the same
+modifiers for each field.  
+
+For example, the following methods throws an exception during encoding:
+
+.. code-block:: java
+   
+   private String getField();
+   public String setField(String x); 
+
+Can I use polymorphism in a POJO accessor?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+No. The codec throws an exception if the types of the accessors do not
+match with the field itself. 
+
+How do I fix: org.bson.codecs.configuration.CodecConfigurationException: Can't find a codec for class X .? 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The class should be public and should have at least one field.
```

Comment:
This means there is no Codec for the class - so a codec needs to be registered for the class.